### PR TITLE
refactor(x/ecocredit): update location to jurisdiction

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- changes usages of location to jurisdiction

Closes: #1012 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/msgs.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func (m *MsgCreateBatch) ValidateBasic() error {
 			}
 
 			if !retiredAmount.IsZero() {
-				if err = ValidateLocation(iss.RetirementLocation); err != nil {
+				if err = ValidateJurisdiction(iss.RetirementLocation); err != nil {
```
Comment (ryanchristo) [inline x/ecocredit/msgs.go:128]: Looks like issuance still needs to be updated for `MsgCreateBatch`.

File: x/ecocredit/msgs.go [hunk 2]
```diff
@@ -182,7 +182,7 @@ func (m *MsgSend) ValidateBasic() error {
 		}
 
 		if !retiredAmount.IsZero() {
-			if err = ValidateLocation(credit.RetirementLocation); err != nil {
+			if err = ValidateJurisdiction(credit.RetirementLocation); err != nil {
```
Comment (ryanchristo) [inline x/ecocredit/msgs.go:185]: Also here for credits within `MsgSend`.

File: x/ecocredit/msgs.go [hunk 3]
```diff
@@ -227,7 +227,7 @@ func (m *MsgRetire) ValidateBasic() error {
 		}
 	}
 
-	if err := ValidateLocation(m.Location); err != nil {
+	if err := ValidateJurisdiction(m.Location); err != nil {
```
Comment (ryanchristo) [inline x/ecocredit/msgs.go:230]: Also within `MsgRetire`.

File: x/ecocredit/msgs.go [hunk 4]
```diff
@@ -506,7 +506,7 @@ func (m *MsgBuy) ValidateBasic() error {
 		}
 
 		if order.RetirementLocation != "" {
-			if err := ValidateLocation(order.RetirementLocation); err != nil {
+			if err := ValidateJurisdiction(order.RetirementLocation); err != nil {
```
Comment (ryanchristo) [inline x/ecocredit/msgs.go:509]: Also within `MsgBuy`.

File: x/ecocredit/msgs.go [hunk 5]
```diff
@@ -578,7 +578,7 @@ func (m *MsgCreateProject) ValidateBasic() error {
 		return ErrMaxLimit.Wrap("create project metadata")
 	}
 
-	if err := ValidateLocation(m.ProjectLocation); err != nil {
+	if err := ValidateJurisdiction(m.ProjectLocation); err != nil {
```
Comment (ryanchristo) [inline x/ecocredit/msgs.go:581]: And here so for `MsgCreateProject`.

File: x/ecocredit/server/core/send.go [hunk 6]
```diff
@@ -151,7 +151,7 @@ func (k Keeper) sendEcocredits(ctx context.Context, credit *core.MsgSend_SendCre
 			Retirer:    to.String(),
 			BatchDenom: credit.BatchDenom,
 			Amount:     sendAmtRetired.String(),
-			Location:   credit.RetirementLocation,
+			Location:   credit.RetirementJurisdiction,
```
Comment (ryanchristo) [inline x/ecocredit/server/core/send.go:154]: And here for `EventRetire`.

**Review by ryanchristo — COMMENTED** [review]
Looks like `Location` is still showing up in a few places.

File: x/ecocredit/msgs.go (hunk 1 above)
Comment (technicallyty) [reply x/ecocredit/msgs.go:128]: i dont think its really necessary to update these, it's pretty much just waiting to be deleted at this point. all the messages have already been migrated to `core/` 🤷🏻

File: x/ecocredit/server/core/send.go (hunk 6 above)
Comment (technicallyty) [reply x/ecocredit/server/core/send.go:154]: ah, looks looks like this was using an outdated event too! good catch 👍🏻

File: x/ecocredit/msgs.go (hunk 1 above)
Comment (ryanchristo) [reply x/ecocredit/msgs.go:128]: Yup, that makes sense.

File: x/ecocredit/server/core/send.go (hunk 6 above)
Comment (ryanchristo) [reply x/ecocredit/server/core/send.go]: Nice. At least one of my comments was relevant.

**Comment by ryanchristo** [comment]
Also looks like another case of random string used in message validation tests causing a race condition:

https://github.com/regen-network/regen-ledger/runs/6032177203?check_suite_focus=true 

Resolved with #1018 when you have a chance to review.

**Comment by ryanchristo** [comment]
I think some of these should just be `jurisdiction` rather than `project_jurisdiction` or `retirement_jurisdiction`.

I think it would be safe to change the following:

- `ProjectInfo` should  use `jurisdiction` rather than `project_jurisdiction`.
- `MsgCreateProject` should use `jurisdiction` rather than `project_jurisdiction`.

I added #1036 to track other changes that we may want to consider but the two changes listed might be nice to include within this pull request.

File: proto/regen/ecocredit/v1/tx.proto [hunk 7]
```diff
@@ -114,13 +114,13 @@ message MsgCreateProject {
...
   string metadata = 3;
 
-  // project_location is the location of the project backing the credits in this
-  // batch. It is a string of the form
+  // project_jurisdiction is the jurisdiction of the project backing the credits
+  // in this batch. It is a string of the form
   // <country-code>[-<sub-national-code>[ <postal-code>]], with the first two
   // fields conforming to ISO 3166-2, and postal-code being up to 64
   // alphanumeric characters. country-code is required, while sub-national-code
   // and postal-code can be added for increasing precision.
```
Comment (aaronc) [inline proto/regen/ecocredit/v1/tx.proto:122]:
```suggestion
  // and postal-code are optional and can be added for increased precision.
```

File: proto/regen/ecocredit/v1/tx.proto [hunk 8]
```diff
@@ -273,13 +272,12 @@ message MsgSend {
...
-    // is a string of the form
-    // <country-code>[-<sub-national-code>[ <postal-code>]], with the first two
-    // fields conforming to ISO 3166-2, and postal-code being up to 64
-    // alphanumeric characters.
-    string retirement_location = 4;
+    // retirement_jurisdiction is the jurisdiction of the beneficiary or buyer
+    // of the retired credits. This must be provided if retired_amount is
+    // positive. It is a string of the form <country-code>[-<sub-national-code>[
+    // <postal-code>]], with the first two fields conforming to ISO 3166-2, and
+    // postal-code being up to 64 alphanumeric characters.
```
Comment (aaronc) [inline proto/regen/ecocredit/v1/tx.proto:279]: It's maybe not a good idea to duplicate all these docs

**Review by aaronc — APPROVED** [review]
Reviewed API changes primarily and those LGTM

**Comment by ryanchristo** [comment]
> I think some of these should just be jurisdiction rather than project_jurisdiction or retirement_jurisdiction. I think it would be safe to change the following:
>
> ProjectInfo should use jurisdiction rather than project_jurisdiction.
> MsgCreateProject should use jurisdiction rather than project_jurisdiction.
>
> I added https://github.com/regen-network/regen-ledger/issues/1036 to track other changes that we may want to consider but the two changes listed above might be nice to include within this pull request.

It looks like the above changes are the only changes necessary needed for #1036 and would have been nice to include here. Also would have been nice to address the proto documentation suggestion aaron made. Let's make sure are are addressing suggestions before merging in the future.

**Pull request merged**